shell/platform/android/io/flutter/view/AccessibilityViewEmbedder.java

```diff
@@ -0,0 +1,312 @@
+        AccessibilityNodeInfo result = AccessibilityNodeInfo.obtain(rootAccessibilityView, flutterId);
+        result.setPackageName(rootAccessibilityView.getContext().getPackageName());
+        result.setSource(rootAccessibilityView, flutterId);
+        result.setClassName(originNode.getClassName());
+
+        Rect displayBounds = flutterIdToDisplayBounds.get(flutterId);
+
+        copyAccessibilityFields(originNode, result);
+        translateBounds(originNode, displayBounds, result);
+        addChildren(originNode, embeddedView, displayBounds, result);
+        setParent(originNode, embeddedView, result);
+
+        return result;
+    }
+
+    private void setParent(AccessibilityNodeInfo originNode, View embeddedView, AccessibilityNodeInfo result) {
+        Long parentOriginPackedId = reflectionAccessors.getParentNodeId(originNode);
+        if (parentOriginPackedId == null) {
+            return;
+        }
+        int parentOriginId = ReflectionAccessors.getVirtualNodeId(parentOriginPackedId);
+        int parentFlutterId = originToFlutterId.get(new ViewAndId(embeddedView, parentOriginId));
+        result.setParent(rootAccessibilityView, parentFlutterId);
+    }
+
+
+    private void addChildren(AccessibilityNodeInfo originNode, View embeddedView, Rect displayBounds, AccessibilityNodeInfo resultNode) {
+        for (int i = 0; i < originNode.getChildCount(); i++) {
+            Long originPackedId = reflectionAccessors.getChildId(originNode, i);
+            if (originPackedId == null) {
+                continue;
+            }
+            int originId = ReflectionAccessors.getVirtualNodeId(originPackedId);
+            ViewAndId origin = new ViewAndId(embeddedView, originId);
+            int childFlutterId;
+            if(originToFlutterId.containsKey(origin)) {
+                childFlutterId = originToFlutterId.get(origin);
+            } else {
+                childFlutterId = nextFlutterId++;
+                originToFlutterId.put(origin, childFlutterId);
+                flutterIdToOrigin.put(childFlutterId, origin);
+                flutterIdToDisplayBounds.put(childFlutterId, displayBounds);
+            }
+            resultNode.addChild(rootAccessibilityView, childFlutterId);
+        }
+    }
+
+    private void translateBounds(AccessibilityNodeInfo originNode, Rect displayBounds, AccessibilityNodeInfo resultNode) {
+        Rect boundsInParent = new Rect();
+        originNode.getBoundsInParent(boundsInParent);
+        resultNode.setBoundsInParent(boundsInParent);
+
+        Rect boundsInScreen = new Rect();
+        originNode.getBoundsInScreen(boundsInScreen);
+        boundsInScreen.offset(displayBounds.left, displayBounds.top);
+        resultNode.setBoundsInScreen(boundsInScreen);
+    }
+
+    private void copyAccessibilityFields(AccessibilityNodeInfo input, AccessibilityNodeInfo output) {
+        output.setAccessibilityFocused(input.isAccessibilityFocused());
```

Comment:
Do we have a good way to keep these in sync? I guess they don't change frequently, but maybe a note like "Please update me when android version bumps?"

Reply:
Well right now this entire solution stops working in Android P, we are still working on a solution that does not require reflection access and or copying the a11y tree. It will take longer for that solution but I suspect once we move to it we won't have to worry about this, and before we move to it, there isn't really a point of worrying about this...

Reply:
Sounds good to me